keps/sig-storage/20191125-bucket-provisioning.md

```diff
@@ -492,6 +483,32 @@ These are the common steps to delete a `Bucket`. Note: there are atypical workfl
 + If the release policy is "Retain" then the `Bucket` remains "Released" and it can potentially be reused.
 + When the COSI controller sees the `Bucket`'s `conditions[x].Reason is "Deleted", it deletes all the finalizers and the real deletions occur.
 
+###  Setting Access Permissions
+#### Dynamic Provisioning
+Incoming BucketAccessRequest either contains a ServiceAccountName where cloud provider supports identity integration or a accessSecretName. In both cases, the incoming BAR representing a request for a user to access the `Bucket`.
+When requesting access to a bucket, workloads will go through the following  scenarios described here:
++  New User and New/Existing Bucket: In this scenario, we do not have user account in the backend storage system as well as no access for this user to the `Bucket`. 
+	+ Create user account in the backend storage system 
+	+ add the access privileges for the user to the `Bucket`
+	+ return the credentials to the workload owning the BAR
++  Existing User and New Bucket: In this scenario, we have the user account already created and exists in the backend storage system but the account is not associated to the `Bucket`. 
+	+ add the access privileges to the `Bucket`
+	+ return the credentials to the workload owning the BAR
++  Existing User and Existing Bucket: In this scenario, we have a user account that has privileges defined on the `Bucket`.  The existing privileges that the user have may conflict with the privileges user requesting. 
+	+ FAIL, if existing privileges are different from the requested privileges
+	+ if the existing privileges match the requested privileges, return the credentials to the workload owning the BAR
++ A Service Account specified and cloud platform identity integration maps Kubernetes ServiceAccounts to the account in the backend system. No need to create credentials here.
+	
+Upon success, BucketAccess Object is created with one of the fields `serviceAccountName` or `accessSecretName` is populated for workloads to use the bucket.
```

Comment:
Also, we have `BucketAccessRequest.accessSecretName` and `BucketAccess.accessSecretName` and they are _not_ the same (at least not according to the KEP. So, either they are really the same, or we should consider changing one of the field names so that they are not confused as being the same. Thoughts?

Reply:
I think I do not want to reiterate the difference between these two fields again.